libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3286,18 +3286,33 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 			(memberName == "min" ||	memberName == "max")
 		)
 			annotation.isPure = true;
-		else if (magicType->kind() == MagicType::Kind::Block && memberName == "chainid" && !m_evmVersion.hasChainID())
+		else if (magicType->kind() == MagicType::Kind::Block)
+		{
+			if (memberName == "chainid" && !m_evmVersion.hasChainID())
 			m_errorReporter.typeError(
 				3081_error,
 				_memberAccess.location(),
 				"\"chainid\" is not supported by the VM version."
 			);
-		else if (magicType->kind() == MagicType::Kind::Block && memberName == "basefee" && !m_evmVersion.hasBaseFee())
-			m_errorReporter.typeError(
-				5921_error,
-				_memberAccess.location(),
-				"\"basefee\" is not supported by the VM version."
-			);
+			else if (memberName == "basefee" && !m_evmVersion.hasBaseFee())
+				m_errorReporter.typeError(
+					5921_error,
+					_memberAccess.location(),
+					"\"basefee\" is not supported by the VM version."
+				);
+			else if (memberName == "difficulty" && m_evmVersion.hasPrevRandao())
+				m_errorReporter.warning(
+					8417_error,
+					_memberAccess.location(),
+					"\"difficulty\" was renamed and supplanted by \"prevrandao\" in the VM version paris."
```

Comment:
Why `renamed and supplanted` and not just `replaced`?

Reply:
It's the terminology they used in the EIP and I believe makes it most clear that it's the same opcode behind it